api/cpp/esp-idf/slint/README.md

```diff
@@ -11,81 +11,110 @@ It has been tested on ESP32-S3 devices.
+[Set up the Environment Variables](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/get-started/linux-macos-setup.html#step-4-set-up-the-environment-variables) instructions.
+* Install a [Rust toolchain for your device](https://esp-rs.github.io/book/installation/index.html).
+* Install the [Rust toolchains for Espressif SoCs](https://esp-rs.github.io/book/installation/index.html).
+  
+### Hello World
 
+The following steps will guide from the a bare-bones esp-idf "hello_world" to a GUI with Slint.
+
+1. Start by creating a new project:
+```bash
+idf.py create-project slint-hello-world
+cd slint-hello-world
+```
+1. Select your chipset with `idf.py set-target`, for example if you're using an `ESP32S3` chipset, run `idf.py set-target esp32s3`
+2. Add a [Board Support Package](https://github.com/espressif/esp-bsp#esp-bsp-espressifs-board-support-packages) that matches your device as a dependency. For example, if you're using an ESP-BOX, run
+```bash
+idf.py add-dependency esp-box
+```
+3. Add Slint as a dependency:
+```bash
+idf.py add-dependency slint/slint
+```
+4. Ensure that Espressif's Rust toolchain is selected for building. Either set the `RUSTUP_TOOLCHAIN` environment variable to the value `esp` or create a file called `rust-toolchain.toml` in your project directory with the following contents:
 ```toml
 [toolchain]
 channel = "esp"
 ```
-
-If you see this error while compiling Slint:
-
+5. Remove `main/slint-hello-world.c`.
+6. Create a new file `main/slint-hello-world.cpp` with the following contents:
+```cpp
+#include <stdio.h>
+#include <esp_err.h>
+#include <bsp/display.h>
+#include <bsp/esp-bsp.h>
+#include <slint-esp.h>
+
+#if defined(BSP_LCD_DRAW_BUFF_SIZE)
+#    define DRAW_BUF_SIZE BSP_LCD_DRAW_BUFF_SIZE
+#else
+#    define DRAW_BUF_SIZE (BSP_LCD_H_RES * CONFIG_BSP_LCD_DRAW_BUF_HEIGHT)
+#endif
+
+#include "appwindow.h"
+
+extern "C" void app_main(void)
+{
+    /* Initialize display  */
+    esp_lcd_panel_io_handle_t io_handle = NULL;
+    esp_lcd_panel_handle_t panel_handle = NULL;
+    const bsp_display_config_t bsp_disp_cfg = {
+        .max_transfer_sz = DRAW_BUF_SIZE * sizeof(uint16_t),
+    };
+    bsp_display_new(&bsp_disp_cfg, &panel_handle, &io_handle);
+
+     /* Set display brightness to 100% */
+    bsp_display_backlight_on();
+
+    std::optional<esp_lcd_touch_handle_t> touch_handle;
+
+    /* Allocate a drawing buffer */
+    static std::vector<slint::platform::Rgb565Pixel> buffer(BSP_LCD_H_RES * BSP_LCD_V_RES);
+
+    /* Initialize Slint's ESP platform support*/
+    slint_esp_init(slint::PhysicalSize({ BSP_LCD_H_RES, BSP_LCD_V_RES }), panel_handle,
+                                       touch_handle, buffer);
+    /* Instantiate the UI */
+    auto ui = AppWindow::create();
+    /* Show it on the screen and run the event loop */
+    ui->run();
+}
 ```
-error: the `-Z` flag is only accepted on the nightly channel of Cargo, but this is the `stable` channel
```

Comment:
IMHO it's worth having a troubleshooting or faq section with this error, because i'm pretty sure users will run into this if they don't read carefully

Reply:
Okay, sounds good. I'll add a section.